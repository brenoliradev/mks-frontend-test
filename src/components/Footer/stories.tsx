import { Story, Meta } from '@storybook/react'

import { Footer } from '.'

export default {
  component: Footer,
  title: 'Footer',
} as Meta

export const Basic: Story = () => <Footer />
