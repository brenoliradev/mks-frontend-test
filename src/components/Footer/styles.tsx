import styled from 'styled-components'

export const FooterWrapper = styled.footer`
  background-color: #eeeeee;
  color: #fff;
  width: 100%;
  height: 34px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  grid-area: footer;
`

export const Text = styled.p`
  color: #000;
  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 15px;
`
