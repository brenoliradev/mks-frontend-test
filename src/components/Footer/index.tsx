import { FooterWrapper, Text } from './styles'

export const Footer = () => {
  return (
    <FooterWrapper>
      <Text role="footertext">MKS sistemas Â© Todos os direitos reservados</Text>
    </FooterWrapper>
  )
}
