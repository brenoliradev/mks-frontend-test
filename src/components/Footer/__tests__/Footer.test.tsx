import { render, screen } from '@testing-library/react'

import Footer from '..'

describe('<Footer />', () => {
  it('should render the footer text', () => {
    render(<Footer />)

    expect(
      screen.getByText(/MKS sistemas Â© Todos os direitos reservados/i),
    ).toBeInTheDocument()
  })

  it('should render p correctly', () => {
    const { container } = render(<Footer />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Footer />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#EEE' })
  })
})
