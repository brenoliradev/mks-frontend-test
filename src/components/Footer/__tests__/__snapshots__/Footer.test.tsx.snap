// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render p correctly 1`] = `
.c0 {
  background-color: #eeeeee;
  color: #fff;
  width: 100%;
  height: 34px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  grid-area: footer;
}

.c1 {
  color: #000;
  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 15px;
}

<footer
  class="c0"
>
  <p
    class="c1"
    role="footertext"
  >
    MKS sistemas Â© Todos os direitos reservados
  </p>
</footer>
`;
